Source/JavaScriptCore/runtime/StructureInlines.h

```diff
@@ -489,7 +490,7 @@ inline PropertyOffset Structure::add(VM& vm, PropertyName propertyName, unsigned
     PropertyOffset newOffset = table->nextOffset(m_inlineCapacity);
 
     m_propertyHash = m_propertyHash ^ rep->existingSymbolAwareHash();
-    m_seenProperties.add(bitwise_cast<uintptr_t>(rep));
+    m_seenProperties.add(WTF::CompactPtr<ClassInfo>::encode(bitwise_cast<uintptr_t>(rep)));
```

Comment:
Why isn't this using the actual type of the pointer, like `WTF::CompactPtr<UniquedStringImpl*>::encode(rep)`?

Ditto for other places like this in the patch.